File: src/main/java/br/com/useinet/finance/dto/TransacaoRequest.java
package br.com.useinet.finance.dto;

import br.com.useinet.finance.model.TipoTransacao;

import java.time.LocalDateTime;

public class TransacaoRequest {
    private String descricao;
    private Double valor;
    private TipoTransacao tipo;
    private LocalDateTime data;
    private Long categoriaId;
    private Long contaId;

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public Double getValor() {
        return valor;
    }

    public void setValor(Double valor) {
        this.valor = valor;
    }

    public TipoTransacao getTipo() {
        return tipo;
    }

    public void setTipo(TipoTransacao tipo) {
        this.tipo = tipo;
    }

    public LocalDateTime getData() {
        return data;
    }

    public void setData(LocalDateTime data) {
        this.data = data;
    }

    public Long getCategoriaId() {
        return categoriaId;
    }

    public void setCategoriaId(Long categoriaId) {
        this.categoriaId = categoriaId;
    }

    public Long getContaId() {
        return contaId;
    }

    public void setContaId(Long contaId) {
        this.contaId = contaId;
    }
}

File: src/main/kotlin/br/com/useinet/finance/dto/TransacaoRequest.kt
package br.com.useinet.finance.dto

import br.com.useinet.finance.model.FrequenciaRecorrencia
import br.com.useinet.finance.model.TipoTransacao
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.time.LocalDate

data class TransacaoRequest(
    @field:NotBlank(message = "Descrição da transação é obrigatória.")
    val descricao: String? = null,

    @field:NotNull(message = "Valor da transação é obrigatório.")
    @field:Positive(message = "Valor da transação deve ser maior que zero.")
    val valor: Double? = null,

    @field:NotNull(message = "Tipo da transação é obrigatório.")
    val tipo: TipoTransacao? = null,

    val data: LocalDate? = null,
    val categoriaId: Long? = null,
    val contaId: Long? = null,
    val recorrente: Boolean = false,
    val frequencia: FrequenciaRecorrencia? = null,
    val proximaOcorrencia: LocalDate? = null
)
